sdk/eventhub/azure-eventhub-checkpointstoretable/azure/eventhub/extensions/checkpointstoretable/_tablestoragecs.py

```diff
@@ -22,17 +27,246 @@ class TableCheckpointStore:
+        :rtype: ~azure.eventhub.extensions.checkpointstoretable.TableCheckpointStore
+        """
+        endpoint, credential = parse_connection_str(
+            conn_str=conn_str, credential=None, keyword_args=kwargs
+        )
+        return cls(endpoint, table_name=table_name, credential=credential, **kwargs)
+
+    def _create_entity_checkpoint(self, checkpoint):
+        my_new_entity = {
+            u'PartitionKey': u'',
+            u'RowKey': u'',
+            u'consumer_group': checkpoint['consumer_group'],
+            u'fully_qualified_namespace': checkpoint['fully_qualified_namespace'],
+            u'eventhub_name': checkpoint['eventhub_name'],
+            u'partition_id': checkpoint['partition_id'],
+            u'offset' : checkpoint['offset'],
+            u'sequence_number' : checkpoint['sequence_number'],
+        }
+        my_new_entity['RowKey'] = my_new_entity['partition_id']
+        my_new_entity['PartitionKey'] = my_new_entity['eventhub_name'] + ' ' + \
+        my_new_entity['fully_qualified_namespace'] + ' ' + my_new_entity['consumer_group'] + ' ' + 'Checkpoint'
+        self.table_client.create_entity(entity=my_new_entity)
+
+    def _create_entity_ownership(self, ownership):
+        my_new_entity = {
+            u'PartitionKey': u'',
+            u'RowKey': u'',
+            u'consumer_group': ownership['consumer_group'],
+            u'fully_qualified_namespace': ownership['fully_qualified_namespace'],
+            u'eventhub_name': ownership['eventhub_name'],
+            u'partition_id':  ownership['partition_id'],
+            u'owner_id' : ownership['owner_id'],
+        }
+        my_new_entity['RowKey'] = my_new_entity['partition_id']
+        my_new_entity['PartitionKey'] = my_new_entity['eventhub_name'] + ' ' \
+        + my_new_entity['fully_qualified_namespace'] + ' ' + my_new_entity['consumer_group'] + ' ' + 'Ownership'
+        new_entity = self.table_client.create_entity(entity=my_new_entity)
+        return new_entity
+
+    @classmethod
+    def _modify_entity_ownership(cls, ownership):
+        """
+        create a dictionary with the new ownership attributes so that it can be updated in tables
+        """
+        my_new_entity = {
+            u'PartitionKey': u'',
+            u'RowKey': u'',
+            u'consumer_group': ownership['consumer_group'],
+            u'fully_qualified_namespace': ownership['fully_qualified_namespace'],
+            u'eventhub_name': ownership['eventhub_name'],
+            u'partition_id':  ownership['partition_id'],
+            u'owner_id' : ownership['owner_id'],
+        }
+        my_new_entity['RowKey'] = my_new_entity['partition_id']
+        my_new_entity['PartitionKey'] = my_new_entity['eventhub_name'] + ' ' \
+        + my_new_entity['fully_qualified_namespace'] + ' ' + my_new_entity['consumer_group'] + ' ' + 'Ownership'
+        return my_new_entity
+
+    @classmethod
+    def _modify_entity_checkpoint(cls, checkpoint):
```

Comment:
`_modify_entity_checkpoint` sounds like it is doing an in-place mutation of the checkpoint, but you're creating a new checkpoint entity.

How about `_create_checkpoint_entity` instead?